File: tests/machine/x/java/match_expr.java
import java.util.*;
import java.util.function.*;
public class MatchExpr {
	public static void main(String[] args) {
	int x = 2;
	Object label = (new java.util.function.Supplier<String>(){public String get(){
	var _t0 = x;
	if (Objects.equals(_t0, 1)) return "one";
	else if (Objects.equals(_t0, 2)) return "two";
	else if (Objects.equals(_t0, 3)) return "three";
	return "unknown";
}}).get();
	System.out.println(label);
	}
}

File: tests/machine/x/cpp/match_expr.cpp
#include <iostream>
#include <string>

int main() {
  auto x = 2;
  auto label = ([&]() {
    auto __v = x;
    if (__v == 1)
      return std::string("one");
    else if (__v == 2)
      return std::string("two");
    else if (__v == 3)
      return std::string("three");
    return std::string("unknown");
  })();
  std::cout << std::boolalpha << label << std::endl;
  return 0;
}
